src/Previewer.js

```diff
@@ -28,9 +30,12 @@ export default function Preview({
 
   let css_provider = null;
   let object_root = null;
+  let language = null;
+  let dbus_proxy = null;
+  let subprocess = null;
 
   preview_window_button.connect("clicked", () => {
-    if (!object_root) return;
+    if (!object_root || language !== "JavaScript") return;
```

Comment:
wdyt of not opening the preview directly and instead showing the same thing as when previewing `GtkWindow` ?

![image](https://user-images.githubusercontent.com/19673/168449626-01a5f266-5d29-4729-8db6-72c05fb4fdb1.png)

This would solve 2 things:

1. Opening the window when loading the demo or clicking on run is a bit abrupt and confusing
2. There are currently 2 live previews

Reply:
I have not solved the case where you have a GtkRoot in the xml. At the moment that does not work, I think.

Though I think having a preview in-window is pretty convenient, even if you still have the external window open, but somewhere in the background. The in-window preview could also only appear when closing the external window though.

And what should "show preview window" do? running can only happen when the preview is visible.

Reply:
> I have not solved the case where you have a GtkRoot in the xml. At the moment that does not work, I think.

That's independent - my suggestion would apply for any kind of object

> Though I think having a preview in-window is pretty convenient

It's confusing and error-prone as the code will only affect one of them. It's really not an option as is.

> The in-window preview could also only appear when closing the external window though.

That's an interesting alternative.

Basically – opening the preview as window would only happen when clicking on "Run". When closiing the preview window - you can resume editing the XML and click on Run again to see the "end result".

I like it because it would simplify the implementation, as you wouldn't need to support real time updates on the out-of-process preview.

Is there a downside to it?

Reply:
> And since you have to click run after updating the preview anyway – is there a downside to it?

Yeah - when previewing a window - you now have 2 windows instead of 1.

That might get confusing